internal/mode/static/manager.go

```diff
@@ -225,25 +226,51 @@ func StartManager(cfg config.Config) error {
 	objects, objectLists := prepareFirstEventBatchPreparerArgs(cfg.GatewayClassName, cfg.GatewayNsName)
 	firstBatchPreparer := events.NewFirstEventBatchPreparerImpl(mgr.GetCache(), objects, objectLists)
 
+	// Ensure NGINX is running before registering metrics & starting the manager.
+	if err := ngxruntime.EnsureNginxRunning(ctx); err != nil {
+		return fmt.Errorf("NGINX is not running: %w", err)
+	}
+
+	if cfg.MetricsConfig.Enabled {
+		if err := configureNginxMetrics(cfg.GatewayClassName); err != nil {
+			return err
+		}
+	}
+
 	eventLoop := events.NewEventLoop(
 		eventCh,
 		cfg.Logger.WithName("eventLoop"),
 		eventHandler,
-		firstBatchPreparer)
+		firstBatchPreparer,
+	)
 
 	if err = mgr.Add(eventLoop); err != nil {
 		return fmt.Errorf("cannot register event loop: %w", err)
 	}
 
-	// Ensure NGINX is running before registering metrics & starting the manager.
-	if err := ngxruntime.EnsureNginxRunning(ctx); err != nil {
-		return fmt.Errorf("NGINX is not running: %w", err)
-	}
-
-	if cfg.MetricsConfig.Enabled {
-		if err := configureNginxMetrics(cfg.GatewayClassName); err != nil {
+	if cfg.LeaderElection.Enabled {
+		leaderElector, err := newLeaderElector(leaderElectorConfig{
+			kubeConfig: clusterCfg,
+			recorder:   recorder,
+			onStartedLeading: func(ctx context.Context) {
+				logger.Info("Started leading")
+				statusUpdater.WriteLastStatuses(ctx)
+			},
+			onStoppedLeading: func() {
+				logger.Info("Stopped leading")
+			},
+			lockNs:   cfg.Namespace,
+			lockName: cfg.LeaderElection.LockName,
+			identity: cfg.LeaderElection.Identity,
+		})
+		if err != nil {
 			return err
 		}
+
+		statusUpdater.SetLeaderElector(leaderElector)
+		go func() {
```

Comment:
I wonder if the controller runtime manager should be responsible for running it all our go routines?

For example, if the manager exists for some reason other than ctx being canceled, will the leader elector go routine be cancelled?